storage/replica.go

```diff
@@ -3177,6 +3187,11 @@ func (r *Replica) executeBatch(
 		optimizePuts(batch, ba.Requests, ba.Header.DistinctSpans)
 	}
 
+	if err := r.checkBatchRange(ba); err != nil {
+		return nil, nil, roachpb.NewErrorWithTxn(err, ba.Header.Txn)
```

Comment:
Hmm, I guess it doesn't hurt. In theory, the txn could've picked up some update and save a retry by being communicated back.